modules/invenio-oaiserver/invenio_oaiserver/response.py

```diff
@@ -346,25 +347,15 @@ def is_private_index(record):
                             etree.QName(NS_OAIPMH, 'metadata'))
 
     etree_record = copy.deepcopy(record)
+    if not etree_record.get('system_identifier_doi', None):
```

Comment:
In case `etree_record` there is no `system_identifier_doi`, it will return None. `Dict.get(key, default=None)`
Please remove None value.